java/registry/src/main/java/io/opensaber/registry/helper/RegistryHelper.java

```diff
@@ -50,10 +63,10 @@ public String addEntity(JsonNode inputJson, String userId) throws Exception {
         String jsonString = objectMapper.writeValueAsString(inputJson);
         try {
             logger.info("Add api: entity type and shard propery: {}", shardManager.getShardProperty());
-
-            Shard shard = shardManager.getShard(inputJson.get(entityType).get(shardManager.getShardProperty()));
+            Map<String, Object> shardIdValue = objectMapper.convertValue(inputJson.get(entityType),Map.class);
```

Comment:
Why a map is needed? Can't it be simple like this
String shardId = inputJson.get(entityType).get(shardManager.getShardProperty());

Reply:
Here it is not considered about shardid, tried to convert to map and get shard from the map(from map default type is integer) but when we try to get from JsonNode it will IntJson.
For this change we modified getShard() in ShardAdvisor.